### ISSUE TITLE
Dots navigation does not work when slidesToShow and slidesToScroll are equal and the number of slides does not divide into slidesToShow evenly

### ISSUE THREAD

**Issue by choppsta** [kenwheeler/slick#1069]
You can recreate the issue on the demo page in the "Add & remove" example.  The example has slidesToShow = 3 and slidesToScroll = 3.  If you add 6 or 9 slides the dots work fine.  If you have 5 or 8 slides the dots no longer work.

I've tracked down the source of the problem to Slick.prototype.getNavigableIndexes() but I've not been able to work out why it's not working.

**Comment by yesdevnull** [comment]
After a lot of playing around I've managed to get it working.  Your hunch about the issue being in [Slick.prototype.getNavigableIndexes()](https://github.com/kenwheeler/slick/blob/master/slick/slick.js#L898) was bang on!

### tl;dr quick fix
```js
counter = (_.slideCount * -1) -1; // Line 907
```

### Details
I have a carousel with 14 items, and 3 items per slide.  When clicking the nav dots, they would always be one dot behind.  I went into [slideHandler](https://github.com/kenwheeler/slick/blob/master/slick/slick.js#L1719) and put in `console.log('slideHandler: ' + index);`.  In the Console, I'd see:

```
slideHandler index: 0
slideHandler index: 3
slideHandler index: 6
```

And so on and so forth, you get the picture.  However, a `console.log(navigables)` (on [Line 641](https://github.com/kenwheeler/slick/blob/master/slick/slick.js#L641)) gets this result:

```
[-14, -11, -8, -5, -2, 1, 4, 7, 10, 13, 16, 19, 22, 25]
```

That doesn't look right, the indexes are off by 1.  The function that populates `navigables` is [Slick.prototype.getNavigableIndexes](https://github.com/kenwheeler/slick/blob/master/slick/slick.js#L898).  

The fix for me was to change `counter` to have -1 more than the total slides.  For example, `14 * -1` is `-14`, which would make the dots click 1 index behind.  By changing the calculation to `(14 * -1) -1`, you now get `-15`.  Now, my math skills aren't particularly strong, but I think the issue is because arrays start at 0.

```js
counter = (_.slideCount * -1) -1; // Line 907
```

Now, a log of the `navigables` array shows:

```
[-14, -12, -9, -6, -3, 0, 3, 6, 9, 12, 15, 18, 21, 24, 27]
```

Winner!  We've got the zero index, and it's incrementing correctly.  [I've made a JSBin example](http://jsbin.com/tajoci/3/) so you can see the different between the old/fixed function and compare the indexes.

I haven't yet had the time to test this extensively, but I haven't run into any issues thus far.

**Comment by yesdevnull** [comment]
I've done some more testing and it appears to be a lot more edge case than I had realised.  My fix above doesn't work correctly for carousels with an even number of slides *sigh*.  Doing more testing now, will update shortly.

**Comment by yesdevnull** [comment]
Righty'o, my commit above should fix it I hope.  There's still a possibility that it may not work correctly if slidesToScroll is a factor of totalSlides.  Will hopefully have time to check it further tonight to make sure it's all working correctly.

```
if ( ( ( _.slideCount % 2 ) == 1 ) || ( ( _.options.slidesToScroll % 2 ) == 1 ) ) {
    counter = (_.slideCount * -1) -1;
} else {
    counter = _.slideCount * -1;
}
```
